### DISCUSSION TITLE
Assigning empty dict to Mapping derived class

### DISCUSSION THREAD

**Discussion by AntoineD** [python/typing#1670] [Q&A]
This code is OK for mypy
```python
from collections.abc import Mapping
MyMapping=Mapping[str, str]
# class MyMapping(Mapping[str, str]): ...
x: MyMapping = {}
```

but this one is not
```python
from collections.abc import Mapping
# MyMapping=Mapping[str, str]
class MyMapping(Mapping[str, str]): ...
x: MyMapping = {}
```

```shell
mypy --strict foo.py
foo.py:4: error: Incompatible types in assignment (expression has type "dict[Never, Never]", variable has type "MyMapping")  [assignment]
```

Could someone explain why and how to deal with it?

**Comment by erictraut** [comment]
Mypy is correct here. The expression `{}` constructs an instance of `dict`. The `dict` class is a subclass of (i.e. derives from) the `Mapping` class. It's not a subclass of `MyMapping`.

If you want to create an instance of `MyMapping`, you'd need to call its constructor rather than using the built-in syntax `{}`.

**Reply by AntoineD** [reply]
Thank you, I feel dumb, that `Never` thing puzzled me.
